share/wake/lib/core/json.wake

```diff
@@ -169,3 +169,143 @@ export def x ==/ y = match x y
   (JArray a) _ = exists (_ ==/ y) a
   _ (JArray b) = exists (_ ==/ x) b
   _ _ = False
+
+# Simplify a JSON structure according to the rules given.
+# For example, Wake's implementation of JSON is more lenient than the base
+# standards, so if maximum compatibility is desired in situations where some
+# other implementation may be parsing generated output, `normalizeJSONCompat`
+# may be used to target that lowest common denominator.
+#
+# Note that the relevant normalization is applied to each member of a `JArray`
+# or `JObject` *before* the enclosing container. This means that if some
+# normalization fails anywhere in the JSON tree the entire call will fail, even
+# if the specific branch causing the failure would later have been pruned.
+export def normalizeJSONWith (fmt: JSONNormalize) (json: JValue): Result JValue Error =
+    def normalizeJSON (path: String) = match _
+        JString a =
+            getJSONNormalizeString fmt a
+            | rmap JString
+            | addErrorContext path
+        JInteger a =
+            getJSONNormalizeInteger fmt a
+            | rmap JInteger
+            | addErrorContext path
+        JDouble a =
+            getJSONNormalizeDouble fmt a
+            | rmap JDouble
+            | addErrorContext path
+        JBoolean a =
+            getJSONNormalizeBoolean fmt a
+            | rmap JBoolean
+            | addErrorContext path
+        JNull =
+            Pass JNull
+        JObject a =
```

Comment:
> Both this and the JArray cases fail to run normalizeJSON recursively after they apply the local fix-up.

I was originally wanting the functions to be able to control their own iteration, but I originally implemented it incorrectly, and working through the logic, even what I thought was "correct" still could never work. Fixed.